pyogrio/geopandas.py

```diff
@@ -62,9 +61,10 @@ def read_dataframe(
     GeoDataFrame or DataFrame (if no geometry is present)
     """
     try:
-        import pandas as pd
-        import geopandas as gp
-        from geopandas.array import from_wkb
+        with GDALEnv():
+            import pandas as pd
+            import geopandas as gp
+            from geopandas.array import from_wkb
```

Comment:
@brendan-ward do you remember why geopandas also needs to be imported in a GDALEnv context? 
(I understand that we need this for our own _ogr functions, but don't understand that for external packages)

I suppose there might be some interference with the fact that geopandas will try to import fiona? (we could also change that to make that in geopandas a really lazy import only when doing read_file/to_file the first time)